### PR TITLE
chore: fix mongoose 7 compatibility

### PR DESCRIPTION
**Pull request by meabed**
With reference to this issues: fixes https://github.com/graphql-compose/graphql-compose-mongoose/issues/426

- Fixed ObjectID typing
- Removed doc.validate callback as it was deprecated
- Fix tests to support mongoose 7 and older versions
- Added yarn command to test mongoose version 6
- Update dependency
- Update github action versions and permission

I have tested it and released the version for me to use until this is merged here: 
https://npmjs.com/package/graphql-compose-mongoose-extended

### CODE REVIEW

**Comment by meabed** [comment]
@nodkz would you please check 
The CI is passing here: https://github.com/meabed/graphql-compose-mongoose/actions

File: src/resolvers/__tests__/removeById-test.ts [hunk 1]
```diff
@@ -167,7 +168,7 @@ describe('removeById() ->', () => {
       expect(exist?.n).toBe(user.name);
     });
 
-    it('should call `beforeQuery` method with non-executed `query` as arg', async () => {
+    it.only('should call `beforeQuery` method with non-executed `query` as arg', async () => {
```
Comment (nodkz) [inline src/resolvers/__tests__/removeById-test.ts:171]: Could you please remove `.only`? It skips a bunch of tests.

File: .github/workflows/nodejs.yml [hunk 2]
```diff
@@ -35,18 +48,21 @@ jobs:
...
+        uses: actions/setup-node@v3
         with:
           node-version: 14.x
       - name: Install node_modules
         run: yarn install
       - name: Build
         run: yarn build
       - name: Semantic Release (publish to npm)
-        run: yarn semantic-release
+        run: npx semantic-release@19
```
Comment (nodkz) [inline .github/workflows/nodejs.yml:65]: Amazing github actions upgrade 🔥

File: src/resolvers/__tests__/removeById-test.ts (hunk 1 above)
Comment (meabed) [reply src/resolvers/__tests__/removeById-test.ts]: Ugh... yea! removed 👍

**Comment by nodkz** [comment]
@meabed  It was very-very clean! 🔥
Thank you very much for your contribution. 👍

**Comment by nodkz** [comment]
Published to npm as 9.8.0

PS. It's strange that semantic-release didn't write a comment itself.

**Comment by meabed** [comment]
It publish tag.

yea, I could enable the GitHub comment plug-in if you need.

Because I added release config file as v19 changed the commit analyser default.

**Comment by nodkz** [comment]
Yep, it will be amazing to add the comment plug-in. It provides a great status info.

I saw a lot repos where code PR is merged but when it will be published is an open question. And SR comments greatly helps to inform contributors when package publishing appears.

**Comment by meabed** [comment]
Yes sounds good will open a PR shorty

**Pull request merged**